code/modules/clothing/neck/_neck.dm

```diff
@@ -185,26 +185,113 @@
 
 	var/mob/living/carbon/carbon_patient = M
 	var/body_part = parse_zone(user.zone_selected)
+	var/oxy_loss = carbon_patient.getOxyLoss()
 
-	var/heart_strength = span_danger("no")
-	var/lung_strength = span_danger("no")
+	var/heart_strength
+	var/pulse_pressure
 
 	var/obj/item/organ/internal/heart/heart = carbon_patient.get_organ_slot(ORGAN_SLOT_HEART)
 	var/obj/item/organ/internal/lungs/lungs = carbon_patient.get_organ_slot(ORGAN_SLOT_LUNGS)
+	var/obj/item/organ/internal/liver/liver = carbon_patient.get_organ_slot(ORGAN_SLOT_LIVER)
+	var/obj/item/organ/internal/appendix/appendix = carbon_patient.get_organ_slot(ORGAN_SLOT_APPENDIX)
+
+	var/render_list = list()//information will be packaged in a list for clean display to the user
+
+	//determine what specific action we're taking
+	if(body_part == BODY_ZONE_CHEST)//We're listening to their chest
+		user.visible_message(span_notice("[user] places [src] against [carbon_patient]'s [body_part] and listens attentively."), ignored_mobs = user)
+		if(!user.can_hear())
+			to_chat(user, span_notice("You place [src] against [carbon_patient]'s [body_part]. Fat load of good it does you though, since you can't hear"))
+			return
+		else
+			render_list += span_info("You place [src] against [carbon_patient]'s [body_part]:\n")
+
+
+	else if(body_part == BODY_ZONE_PRECISE_GROIN)//We're feeling their abdomen
+		render_list += span_info("You carefully press down on [carbon_patient]'s abdomen:\n")
+		user.visible_message(span_notice("[user] presses their hands against [carbon_patient]'s abdomen."), ignored_mobs = user)
+	else if(body_part == BODY_ZONE_HEAD)//We're feeling their pulse on their neck
+		render_list += span_info("You carefully press your fingers to [carbon_patient]'s neck:\n")
+		user.visible_message(span_notice("[user] presses their fingers against [carbon_patient]'s neck."), ignored_mobs = user)
+	else if(body_part != BODY_ZONE_CHEST && body_part != BODY_ZONE_PRECISE_GROIN && body_part != BODY_ZONE_PRECISE_EYES && body_part != BODY_ZONE_PRECISE_MOUTH)//We're feeling their pulse on an extremity
+		render_list += span_info("You carefully press your fingers to [carbon_patient]'s [body_part]:\n")
+		user.visible_message(span_notice("[user] presses their fingers against [carbon_patient]'s [body_part]."), ignored_mobs = user)
+	else//We're targeting their mouth or eyes. Try again.
+		balloon_alert(user, "can't do that!")
+		return
```

Comment:
Switch statement instead?

https://www.byond.com/docs/ref/index.html#/proc/switch

Reply:
So that's how switch statements work in DM, huh. Cheers, implemented!